### DISCUSSION TITLE
Enable Panning Left/Right on Waterfall When Zoomed in on Mobile Browsers

### DISCUSSION THREAD

**Discussion by bhall7** [eroyee/openwebrx_E#4] [Ideas]
On mobile web browsers, the user is unable to pan left or right on the waterfall when zoomed in. On traditional web browsers, the user can easily pan left or right by simply clicking and dragging. But, in OpenWebRX on a mobile browser, one must zoom all the way out first, then tap to tune the desired area, and then zoom in.

It would be great if OpenWebRX would allow scrolling/panning left/right on mobile like KiwiSDR does. Or, borrowing another KiwiSDR feature would be to optionally have the pan left/right buttons on the receiver panel.

This is such an important capability while tuning through the band hunting for signals.

**Comment by eroyee** [comment]
I understand how this function would be useful.

Unfortunately I've no mobile system here at all (happily using a 'brick' 'phone :) so I can't test for the problem myself, however I did have a quick look at a KiwiSDR and wasn't able to find the pan buttons. Most probably that's user error (me) so I'll need to check again more thoroughly.

In terms of 'fixing' it, I suspect the best approach would be to test for a mobile browser and present a different view (maybe KiwiSDR does that already, and why I didn't get these buttons?). In terms of my own 'roadmap' this matters because I'd be wary of adding even more buttons to the standard receiver panel over what I've already done.

Even though it probably won't be me that's able to improve this, for the sake of discussion and hope that someone with development skills and a mobile browser comes along, I wonder if perhaps a floating arrow at each extremity of the waterfall might work? That would seem to be reasonably intuitive and would not add to button congestion on the receiver panel?

**Reply by bhall7** [reply]
You can use the Responsive Design Mode in Firefox (https://firefox-source-docs.mozilla.org/devtools-user/responsive_design_mode/ ; Ctrl + Shift + M) and in all other modern browsers to emulate a mobile device. You can get to it from the Developer Tools menu, or directly using a shortcut similar to bove.

In KiwiSDR, the "pan" buttons are bounded here in red:
![image](https://user-images.githubusercontent.com/513487/176216837-f5f7006d-49dc-4df9-9b64-79608f476a17.png)

And yes, a floating arrow at each extremity of the waterfall would also be acceptable.

**Comment by eroyee** [comment]
Huh, well I certainly saw those buttons but some some reason they didn't do anything on the unit I tried. 

Ah well, as I said it must have been user error 'cos I just had another go and they worked this time!

**Reply by bhall7** [reply]
Cool, yeah, you have to be zoomed in for the L/R pan buttons to work. Not sure if that’s why it wasn’t working before. But, glad it’s working now. Thanks for your consideration.

**Comment by eroyee** [comment]
I've just been working on this...

... could you update your system and try it out with your mobile please?

**Reply by bhall7** [reply]
What files do I need to copy, and do I need to recompile or reinstall any other components?

**Reply by bhall7** [reply]
Never mind...I had a look at the last commit and manually copied the changed files. Feature works perfectly! Thank you SO much! This is exactly what I've been needing for a long time now. 🙌

**Comment by eroyee** [comment]
Excellent, pleased to hear it - thanks for letting me know.

**Comment by eroyee** [comment]
Incidentally (I meant to mention this earlier), I think the issue I had with the KiwiSDR was that hovering over the pan buttons simply said something like "page down alt:label step", which was rather non-intuitive as to what they did.

I'm not sure if this is intentional or not, but I'll mention it to @jks-prv in case.
